src/content/en/updates/2019/08/get-started-with-gpu-compute-on-the-web.md

```diff
@@ -384,9 +384,9 @@ const shaderModule = device.createShaderModule({
       numbers: array<f32>;
     };
 
-    [[group(0), binding(0)]] var<storage> firstMatrix : [[access(read)]] Matrix;
-    [[group(0), binding(1)]] var<storage> secondMatrix : [[access(read)]] Matrix;
-    [[group(0), binding(2)]] var<storage> resultMatrix : [[access(write)]] Matrix;
+    [[group(0), binding(0)]] var<storage, read> firstMatrix : Matrix;
+    [[group(0), binding(1)]] var<storage, read> secondMatrix : Matrix;
+    [[group(0), binding(2)]] var<storage, write> resultMatrix : Matrix;
 
     [[stage(compute)]] fn main([[builtin(global_invocation_id)]] global_id : vec3<u32>) {
       resultMatrix.size = vec2<f32>(firstMatrix.size.x, secondMatrix.size.y);
```

Comment:
nit: in the code below you should be able to remove `var` and `let` type annotations:

```
    [[block]] struct Matrix {
      size : vec2<f32>;
      numbers: array<f32>;
    };

    [[group(0), binding(0)]] var<storage, read> firstMatrix : Matrix;
    [[group(0), binding(1)]] var<storage, read> secondMatrix : Matrix;
    [[group(0), binding(2)]] var<storage, write> resultMatrix : Matrix;

    [[stage(compute)]] fn main([[builtin(global_invocation_id)]] global_id : vec3<u32>) {
      resultMatrix.size = vec2<f32>(firstMatrix.size.x, secondMatrix.size.y);

      let resultCell = vec2<u32>(global_id.x, global_id.y);
      var result = 0.0;
      for (var i = 0u; i < u32(firstMatrix.size.y); i = i + 1u) {
        let a = i + resultCell.x * u32(firstMatrix.size.y);
        let b = resultCell.y + i * u32(secondMatrix.size.y);
        result = result + firstMatrix.numbers[a] * secondMatrix.numbers[b];
      }

      let index = resultCell.y + resultCell.x * u32(secondMatrix.size.y);
      resultMatrix.numbers[index] = result;
    }
```

Reply:
I'll do that when Chrome Canary supports those. Thanks!